django/db/models/fields/related_descriptors.py

```diff
@@ -172,23 +171,26 @@ def __get__(self, instance, cls=None):
         if instance is None:
             return self
 
-        # The related instance is loaded from the database and then cached in
-        # the attribute defined in self.cache_name. It can also be pre-cached
+        # The related instance is loaded from the database and then cached
+        # by the field on the model instance state. It can also be pre-cached
         # by the reverse accessor (ReverseOneToOneDescriptor).
         try:
-            rel_obj = getattr(instance, self.cache_name)
-        except AttributeError:
+            rel_obj = self.field.get_cached_value(instance)
```

Comment:
Yeah we don't want that. If we did modify this to use a default value instead of raising, it'd need to be a unique sentinel default, not None. I don't feel strongly that that's an improvement, though.